File: atcoder/abc155B_papers_please.py
# unihernandez22
# https://atcoder.jp/contests/abc155/tasks/abc155_b
# implementation

input()
a = map(int, input().split())
for i in a :
    if i%2==0 and i%3 != 0 and i%5 != 0 :
        print("DENIED")
        break
else:
    print("APPROVED")

File: atcoder/abc155B_papers_please.cpp
// https://atcoder.jp/contests/abc155/tasks/abc155_b
#include<iostream>
#include<vector>

using namespace std;

int main() {
    int n, tmp;
    cin >> n;

    vector<int> even;
    for (int i = 0; i < n; i++) {
        cin >> tmp;
        if (tmp % 2 == 0) even.push_back(tmp);
    }

    for (auto number : even) {
        if (number % 3 != 0 && number % 5 != 0) {
            cout << "DENIED" << endl;
            return 0;
        }
    }
    cout << "APPROVED" << endl;

    return 0;
}
